src/ess/reflectometry/tools.py

```diff
@@ -270,3 +273,37 @@ def combine_curves(
         ),
         coords={'Q': qgrid},
     )
+
+
+def orso_datasets_from_measurements(workflow, files, scale_to_overlap=True):
+    reflectivity_curves = []
+    for fname, quantities in files.items():
+        wf = workflow.copy()
+        wf[Filename[SampleRun]] = fname
+        for name, value in quantities.items():
+            wf[name] = value
+        reflectivity_curves.append(wf.compute(NormalizedIofQ))
+
+    scale_factors = (
+        scale_reflectivity_curves_to_overlap(
+            [r.hist() for r in reflectivity_curves], return_scaling_factors=True
+        )
+        if scale_to_overlap
+        else (1,) * len(files)
+    )
+
+    datasets = []
+    for (fname, quantities), curve, scale_factor in zip(
+        files.items(), reflectivity_curves, scale_factors, strict=True
+    ):
+        wf = workflow.copy()
+        wf[Filename[SampleRun]] = fname
+        for name, value in quantities.items():
+            wf[name] = value
+        wf[NormalizedIofQ] = scale_factor * curve
+        dataset = wf.compute(orso.OrsoIofQDataset)
```

Comment:
```python
datasets = sciline.compute_mapped(wf.map(dataframe), orso.OrsoIofQDataset)
```
If you setup `dataframe` with a `NormalizedIofQ` column (I am not 100% we can map non-source nodes. Edit: No we cannot, I think we should add `__delitem__` as also discussed  in scipp/cyclebane#14).
Not sure about the `find_corrections`, but I presume they are the same for all files, i.e., you can just run that once on the base workflow?

Reply:
> but I presume they are the same for all files, i.e., you can just run that once on the base workflow?

They are probably the same, but I don't see the benefit of making that assumption here.

Reply:
You are literally using the same workflow, how could it be different?

Reply:
What corrections are applied could depend on the parameters of the workflow. Right now they don't but I don't think that is an implausible situation.

Reply:
Seems quite implausible, I thought this is for handling measurements from the same "run" at multiple angles?

But if you are convinced that this is what you need, in that case I don't have any suggestions for improvement.

Reply:
> Seems quite implausible, I thought this is for handling measurements from the same "run" at multiple angles?

That's the main use case yes. It's wrong to say that I'm convinced this is what we need. I don't know, and therefore opted to keep it flexible, since the cost of doing so is negligible.